client/ayon_core/pipeline/tempdir.py

```diff
@@ -3,11 +3,75 @@
 """
 
 import os
+import tempfile
+from pathlib import Path
+
 from ayon_core.lib import StringTemplate
 from ayon_core.pipeline import Anatomy
 
 
-def create_custom_tempdir(project_name, anatomy=None):
+def get_temp_dir(
+    project_name, anatomy=None, prefix=None, suffix=None, use_local_temp=False
+):
+    """Get temporary dir path.
+
+    If `use_local_temp` is set, tempdir will be created in local tempdir.
+    If `anatomy` is not set, default anatomy will be used.
+    If `prefix` or `suffix` is not set, default values will be used.
+
+    It also supports `AYON_TMPDIR`, so studio can define own temp
+    shared repository per project or even per more granular context.
+    Template formatting is supported also with optional keys. Folder is
+    created in case it doesn't exists.
+
+    Args:
+        project_name (str): Name of project.
+        anatomy (Optional[Anatomy]): Project Anatomy object.
+        suffix (Optional[str]): Suffix for tempdir.
+        prefix (Optional[str]): Prefix for tempdir.
+        use_local_temp (Optional[bool]): If True, temp dir will be created in
+            local tempdir.
+
+    Returns:
+        str: Path to staging dir of instance.
+
+    """
+    prefix = prefix or "ay_tmp_"
+    suffix = suffix or ""
+
+    if use_local_temp:
+        return _create_local_staging_dir(prefix, suffix)
+
+    # make sure anatomy is set
+    if not anatomy:
+        anatomy = Anatomy(project_name)
+
+    # get customized tempdir path from `OPENPYPE_TMPDIR` env var
+    custom_temp_dir = _create_custom_tempdir(anatomy.project_name, anatomy)
+
+    return _create_local_staging_dir(prefix, suffix, dirpath=custom_temp_dir)
+
+
+def _create_local_staging_dir(prefix, suffix, dirpath=None):
+    """Create local staging dir
+
+    Args:
+        prefix (str): prefix for tempdir
+        suffix (str): suffix for tempdir
+        dirpath (Optional[str]): path to tempdir
+
+    Returns:
+        str: path to tempdir
+    """
+    # use pathlib for creating tempdir
+    staging_dir = Path(tempfile.mkdtemp(
+        prefix=prefix, suffix=suffix, dir=dirpath
+    ))
+
+    return staging_dir.as_posix()
```

Comment:
Why is the `Path` conversion necessary?
```suggestion
    return tempfile.mkdtemp(
        prefix=prefix, suffix=suffix, dir=dirpath
    )
```

Reply:
This was in the initial PR I picked-up so I left it and assumed it was to conform as posix and avoid potential more problems down the line. But I can see current code does not perform this conform indeed.
Shall I get rid of it ?

Reply:
Yes please.